### ISSUE TITLE
Build error in deepstream 6

### ISSUE THREAD

**Issue by shaojun** [neilyoung/nvdsinfer_custom_impl_onnx#1]
I got this repo from [forums.developer.nvidia.com](https://forums.developer.nvidia.com/t/deepstream-5-1-pytorch-mobilenet-ssd-v1-retained-onnx-poor-performance/175468)

But I'm using DeepStream 6 now, I tried to build the repo on the Jetson Nano, always show an error:

> shao@shao-jsn-4g:/opt/nvidia/deepstream/deepstream/sources/objectDetector_SSD/nvdsinfer_custom_impl_onnx-master$ ls
> Makefile  nvdsiplugin_onnx.cpp  nvdsparsebbox_onnx.cpp  README.md
> shao@shao-jsn-4g:/opt/nvidia/deepstream/deepstream/sources/objectDetector_SSD/nvdsinfer_custom_impl_onnx-master$ make
> g++ -o libnvdsinfer_custom_impl_onnx.so nvdsparsebbox_onnx.cpp nvdsiplugin_onnx.cpp -Wall -Werror -std=c++11 -shared -fPIC -Wno-error=deprecated-declarations -I../../includes -I/usr/local/cuda-10.2/include -Wl,--start-group -lnvinfer -lnvparsers -L/usr/local/cuda-10.2/lib64 -lcudart -lcublas -Wl,--end-group
> nvdsiplugin_onnx.cpp:91:9: error: looser throw specifier for ‘virtual int FlattenConcat::getNbOutputs() const’
>      int getNbOutputs() const override { return 1; }
>          ^~~~~~~~~~~~
> In file included from /usr/include/aarch64-linux-gnu/NvInferLegacyDims.h:53:0,
>                  from /usr/include/aarch64-linux-gnu/NvInfer.h:53,
>                  from /usr/include/aarch64-linux-gnu/NvInferPlugin.h:53,
>                  from nvdsiplugin_onnx.cpp:23:
> /usr/include/aarch64-linux-gnu/NvInferRuntimeCommon.h:447:21: error:   overriding ‘virtual int32_t nvinfer1::IPluginV2::getNbOutputs() const noexcept’
>      virtual int32_t getNbOutputs() const noexcept = 0;
>                      ^~~~~~~~~~~~
> nvdsiplugin_onnx.cpp:93:10: error: looser throw specifier for ‘virtual nvinfer1::Dims FlattenConcat::getOutputDimensions(int, const Dims*, int)’
>      Dims getOutputDimensions(int index, const Dims* inputs, int nbInputDims) override
>           ^~~~~~~~~~~~~~~~~~~
> In file included from /usr/include/aarch64-linux-gnu/NvInferLegacyDims.h:53:0,
>                  from /usr/include/aarch64-linux-gnu/NvInfer.h:53,
>                  from /usr/include/aarch64-linux-gnu/NvInferPlugin.h:53,
>                  from nvdsiplugin_onnx.cpp:23:
> /usr/include/aarch64-linux-gnu/NvInferRuntimeCommon.h:459:18: error:   overriding ‘virtual nvinfer1::Dims nvinfer1::IPluginV2::getOutputDimensions(int32_t, const Dims*, int32_t) noexcept’
>      virtual Dims getOutputDimensions(int32_t index, Dims const* inputs, int32_t nbInputDims) noexcept = 0;
>                   ^~~~~~~~~~~~~~~~~~~
> nvdsiplugin_onnx.cpp:128:9: error: looser throw specifier for ‘virtual int FlattenConcat::initialize()’
>      int initialize() override
>          ^~~~~~~~~~
> In file included from /usr/include/aarch64-linux-gnu/NvInferLegacyDims.h:53:0,
>                  from /usr/include/aarch64-linux-gnu/NvInfer.h:53,
>                  from /usr/include/aarch64-linux-gnu/NvInferPlugin.h:53,
>                  from nvdsiplugin_onnx.cpp:23:
> /usr/include/aarch64-linux-gnu/NvInferRuntimeCommon.h:511:21: error:   overriding ‘virtual int32_t nvinfer1::IPluginV2::initialize() noexcept’
>      virtual int32_t initialize() noexcept = 0;
>                      ^~~~~~~~~~
> nvdsiplugin_onnx.cpp:134:10: error: looser throw specifier for ‘virtual void FlattenConcat::terminate()’
>      void terminate() override
>           ^~~~~~~~~
> In file included from /usr/include/aarch64-linux-gnu/NvInferLegacyDims.h:53:0,
>                  from /usr/include/aarch64-linux-gnu/NvInfer.h:53,
>                  from /usr/include/aarch64-linux-gnu/NvInferPlugin.h:53,
>                  from nvdsiplugin_onnx.cpp:23:
> /usr/include/aarch64-linux-gnu/NvInferRuntimeCommon.h:518:18: error:   overriding ‘virtual void nvinfer1::IPluginV2::terminate() noexcept’
>      virtual void terminate() noexcept = 0;
>                   ^~~~~~~~~
> 

do you have any idea?
Thanks!

**Comment by neilyoung** [comment]
No, sorry. I'm no longer working in this field and the scope of the solution is very narrow.

But it seems to be a common problem https://forums.developer.nvidia.com/t/deepstream-sdk-5-1-objectdetector-ssd-compilation-error/181765
